src/example_fgen_basic/result/result_int.f90

```diff
@@ -0,0 +1,128 @@
+!> Result type for integers
+!>
+!> Inspired by the excellent, MIT licensed
+!> https://github.com/samharrison7/fortran-error-handler
+module m_result_int
+
+    use kind_parameters, only: i8
+    use m_error_v, only: ErrorV
+    use m_result, only: ResultBase
+    use m_result_none, only: ResultNone
+
+    implicit none (type, external)
+    private
+
+    type, extends(ResultBase), public :: ResultInt
+    !! Result type that holds integer values
+
+        integer(kind=i8), allocatable :: data_v
+        !! Data i.e. the result (if no error occurs)
+
+        ! Note: the error_v attribute comes from ResultBase
+
+    contains
+
+        private
+
+        procedure, public :: build
+        procedure, public :: finalise
+        final :: finalise_auto
+
+    end type ResultInt
+
+    interface ResultInt
+    !! Constructor interface - see build [TODO: x-ref] for details
+        module procedure :: constructor
+    end interface ResultInt
+
+contains
+
+    function constructor(data_v, error_v) result(self)
+        !! Build instance
+
+        type(ResultInt) :: self
+        ! Hopefully can leave without docstring (like Python)
+
+        integer(kind=i8), intent(in), optional :: data_v
+        !! Data
+
+        class(ErrorV), intent(in), optional :: error_v
+        !! Error
+
+        type(ResultNone) :: build_res
+
+        build_res = self % build(data_v_in=data_v, error_v_in=error_v)
+
+        if (build_res % is_error()) then
+
+            ! This interface has to return the initialised object,
+            ! it cannot return a Result type,
+            ! so we have no choice but to raise a fatal error here.
+            print *, build_res % error_v % message
+            error stop build_res % error_v % code
+
+        ! else
+            ! Assume no error occurred and initialisation was fine
+
+        end if
+
+    end function constructor
+
+    function build(self, data_v_in, error_v_in) result(res)
+        !! Build instance
```

Comment:
```suggestion
    subroutine build(self, data_v_in, error_v_in, res)
        !! Build instance
```

I'm not sure what I like more. I don't love the pattern of using a function with effectively multiple return values. Now that I've seen it, using a subroutine seems to be a better option. What do you think?